crates/python/src/api.rs

```diff
@@ -102,6 +102,14 @@ py_wrap_error!(
     PyRuntimeError
 );
 
+/// Get the keyword `key` value from `kwds` if it is of type `Option<T>` and it is present, else `None`.
+/// Returns an error if a value is present but cannot be extracted into `T`.
```

Comment:
Doesn't look like this returns an error due to the `unwrap_or(Ok(None))`

Reply:
After the `.map` the type is `Option<Result<Option<T>>` so the `.unrap_or` is unpacking the outer option, which is contingent on `kwds` being `Some(kwds)`

Reply:
Oh, right. Also if it didn't return a result, type checking would fail, so was a derp moment from me :)